lib/backend/k8s/resources/node.go

```
@@ -153,24 +157,48 @@ func (c *nodeClient) List(ctx context.Context, list model.ListInterface, revisio
 		}, nil
 	}
 
-	// Listing all nodes.
-	nodes, err := c.clientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{ResourceVersion: revision})
-	if err != nil {
-		return nil, K8sErrorToCalico(err, list)
+	listFunc := func(ctx context.Context, opts metav1.ListOptions) (runtime.Object, error) {
+		nodes, err := c.clientSet.CoreV1().Nodes().List(ctx, opts)
+		if err != nil {
+			return nil, fmt.Errorf("failed to list nodes: %s", err)
```

Need to return the actual error type here, not wrapped.